### ISSUE TITLE
Unwanted blank lines in PowerShell output in VSCode

### ISSUE THREAD

**Issue by alexandair** [dotnet/interactive#641]
### Describe the bug
Output of the PowerShell cell contains a blank line after every line of the output. Please, fix the formatting of the output.

### Please complete the following:

**Which version of .NET Interactive are you using?** 
![image](https://user-images.githubusercontent.com/3135708/87341221-8998d300-c549-11ea-8b54-aad360a7d384.png)

- OS
    - [x] Windows 10
    - [ ] macOS
    - [ ] Linux (Please specify distro)
    - [ ] iOS
    - [ ] Android
 - Browser 
    - [ ] Chrome
    - [ ] Edge
    - [ ] Firefox
    - [ ] Safari
 - Frontend
    - [ ] Jupyter Notebook
    - [ ] Jupyter Lab
    - [ ] nteract
    - [x] Visual Studio Code
    - [ ] Other (please specify)

### Screenshots
![image](https://user-images.githubusercontent.com/3135708/87341137-5fdfac00-c549-11ea-9c1b-4efc737bec09.png)

**Comment by jonsequitur** [comment]
@TylerLeonhardt Do you know why this happens by any chance?

**Comment by TylerLeonhardt** [comment]
I feel like this is a formatting thing more than a PowerShell specific thing...

Like in a C# 
```
Console.WriteLine(1);
Console.WriteLine(2);
Console.WriteLine(3);
```

you get the same thing.

**Comment by alexandair** [comment]
.NET PowerShell kernel in Azure Data Studio behaves the same and outputs those extra blank lines.
At the same time, "old" PowerShell kernel's output is fine. 

.NET PowerShell kernel

![image](https://user-images.githubusercontent.com/3135708/87772651-5349a500-c822-11ea-8a92-18784ba0ac16.png)

PowerShell kernel
![image](https://user-images.githubusercontent.com/3135708/87772559-33b27c80-c822-11ea-9c01-456c2211fda7.png)

**Comment by jonsequitur** [comment]
It looks like this is because each write to the console is a separate display event.

@rebornix, thoughts?

**Comment by TylerLeonhardt** [comment]
@jonsequitur, @alexandair's example is in Azure Data Studio https://github.com/dotnet/interactive/issues/641#issuecomment-660001271

**Comment by jonsequitur** [comment]
Yeah, I see that. Sorry for not being clearer. I meant to point out that VS Code exhibits the same behavior and presumably for similar reasons, i.e. DOM and styling. You can see from my screen shot that we're not inserting extra newlines. And this doesn't happen in JupyterLab:

![image](https://user-images.githubusercontent.com/547415/88109452-64b2ea00-cb5f-11ea-8204-fb5f16c41a03.png)


Maybe there's a different API we should call when adding these items to the DOM? @brettfo?

**Comment by brettfo** [comment]
The issue is that pwsh is returning the formatted text with a trailing newline, e.g. `text/plain` for each `"1\n"`, `"2\n"`, etc. and the CSS that VS Code applies to the outputs preserves the whitespace (which, arguably it should since it's raw text.)

We _could_ add `.trimEnd()` on the notebook side (or whatever the call is) to all formatted text, but I'm hesitant to do so because it's entirely possible for someone to write this and we'd do the wrong thing trying to to the right thing: `Console.WriteLine("lots of empty space:\n\n\n\n\n"); Console.WriteLine("some text down low");` and that would get rendered as:

```
lots of empty space:
some text down low
```

Ideally the pwsh kernel could detect when it's interesting to trim its own trailing newlines when constructing the formatted text.

**Comment by jonsequitur** [comment]
You know I love Perl.

I think this is a good solution. It'll eventually surprise someone but the current behavior is surprising to everyone.

**Comment by brettfo** [comment]
@jonsequitur

> I think this is a good solution.

1. Removing a single trailing newline if present from **all** formatted text? or
2. Updating the pwsh kernel to trim its own newlines?

**Comment by jonsequitur** [comment]
The same problem exists if you submit this in C#, so option 1 is probably better:

```csharp
Console.WriteLine("1");
Console.WriteLine("2");
```

**Comment by brettfo** [comment]
After experimenting for a bit, we're doing the right thing and it's VS Code that needs to adjust their display/CSS.

I modified the output to not contain trailing newlines and the result was that (1) VS Code still has the extra vertical space, so no change here, and (2) Jupyter Lab started displaying stuff on the same line (e.g., `12345`), so a negative change.  VS Code also shows the same line gaps with `Console.Write`.

I'm closing this issue as it's not something we can directly alter, but I've opened an issue on VS Code to track it: microsoft/vscode#103661.
